apps/aechannel/src/aesc_fsm.erl

```diff
@@ -990,15 +1034,31 @@ half_signed(Type, Msg, D) ->
     handle_common_event(Type, Msg, error_all, D).
 
 mutual_closing(enter, _OldSt, _D) -> keep_state_and_data;
+mutual_closing(cast, {?SHUTDOWN_ERR, Msg}, D) ->
```

Comment:
Possibly related to [previous comment](https://github.com/aeternity/aeternity/pull/2737/files#r320393959) for an attack vector. In this case malicious Bob keeps the close mutual Alice solo-authenticated and Bob can co-authenticate it and post it at any point now. Only protection Alice has is posting a new transaction (of any type) on-chain thus invalidating the `nonce` in the `channel_close_mutual` transaction Bob holds.

Reply:
Although this will not work for the specific case of `generalized` accounts that don't have a meaningful replay-attack protection, I don't think this is a realistic issue

Reply:
> Possibly related to [previous comment](https://github.com/aeternity/aeternity/pull/2737/files#r320393959) for an attack vector. In this case malicious Bob keeps the close mutual Alice solo-authenticated and Bob can co-authenticate it and post it at any point now. Only protection Alice has is posting a new transaction (of any type) on-chain thus invalidating the `nonce` in the `channel_close_mutual` transaction Bob holds.

I'm not sure how to automatically protect against that, but Alice _will_ get notified about the situation and could possibly glean that Bob now may have a co-signed shutdown tx armed and ready. If it appears on-chain later, the watcher will at least detect it.

Reply:
Since Alice was the author of the `channel_close_mutual` transaction, it is Alice who is the `origin` of the transaction. If Alice posts a new transaction on-chain (spend transaction to herself) with the same `nonce` and this gets included, the transaction that Bob now holds will be invalid and he will not be able to post it ever again. This will work with `basic` and `generalized` accounts alike if a replay-attack is present in the GA contract.

Not sure if this is part of the FSM in the first place.